### DISCUSSION TITLE
Vitest not resolving auto-imports

### DISCUSSION THREAD

**Discussion by luukrijnbende** [nuxt/nuxt#16878] [Questions]
Hi All,

I'm trying to setup Vitest to work with Nuxt 3, but I'm running into auto-import issues.
I have a simple server side route which uses `defineEventHandler` which is automatically imported when Nuxt runs, but this is not the case when running Vitest.

```
// server/routes/something.ts
export default defineEventHandler(() => 'OK');
```

```
// test/routes/something.spec.ts
import { it } from 'vitest';
import handler from '../../../server/routes/something';

it('should return OK', () => {
  const response = handler({} as any);

  expect(response).toBe('OK');
});
```

The test fails with the following error.
```
FAIL  src/server/routes/something.spec.ts [ src/server/routes/something.spec.ts ]
ReferenceError: defineEventHandler is not defined
 ❯ src/server/routes/something.ts:1:44
      1| export default defineEventHandler(() => 'OK');
       |                                            ^
```

I created a Stackblitz with a reproduction https://stackblitz.com/edit/nuxt-starter-nvmcae
Run `npm run test` to run the tests.

Does anyone know how to configure Vitest to work with Nuxt 3 smoothly?

**Comment by mitraecp** [comment]
> Hi All,
> 
> I'm trying to setup Vitest to work with Nuxt 3, but I'm running into auto-import issues. I have a simple server side route which uses `defineEventHandler` which is automatically imported when Nuxt runs, but this is not the case when running Vitest.
> 
> ```
> // server/routes/something.ts
> export default defineEventHandler(() => 'OK');
> ```
> 
> ```
> // test/routes/something.spec.ts
> import { it } from 'vitest';
> import handler from '../../../server/routes/something';
> 
> it('should return OK', () => {
>   const response = handler({} as any);
> 
>   expect(response).toBe('OK');
> });
> ```
> 
> The test fails with the following error.
> 
> ```
> FAIL  src/server/routes/something.spec.ts [ src/server/routes/something.spec.ts ]
> ReferenceError: defineEventHandler is not defined
>  ❯ src/server/routes/something.ts:1:44
>       1| export default defineEventHandler(() => 'OK');
>        |                                            ^
> ```
> 
> I created a Stackblitz with a reproduction https://stackblitz.com/edit/nuxt-starter-nvmcae Run `npm run test` to run the tests.
> 
> Does anyone know how to configure Vitest to work with Nuxt 3 smoothly?

you find some about it?

**Comment by anaroblesd** [comment]
Hello, I found a solution for this problem. Found the example here: https://medium.com/@ana.robles.developer/nuxt-3-vitest-vue-test-utils-mockear-en-los-test-funciones-auto-importadas-2242297eb06d
Hope it works for you too.

**Reply by pmwmedia** [reply]
Unfortunately, the linked article does not solve the `defineEventHandler is not defined` issue.
